server/server_test.go

```diff
@@ -831,3 +801,20 @@ func (dh dummyHandler) Loop() {
 func (dh dummyHandler) Stop() {
 	close(dh.quit)
 }
+
+func setRandomPorts(t *testing.T, network string, cfg *Config) {
```

Comment:
[nit] I would clarify it.

```suggestion
func setAutoAssignedPorts(t *testing.T, network string, cfg *Config) {
```

Reply:
Thanks, that's a better name.